src/lib.rs

```diff
@@ -180,7 +186,21 @@ impl<'a, T> SearchTree<T>
 
     fn left_rotation(&mut self, index: usize) {
         if let Some(right_index) = self.nodes[index].right_child {
+            // The right child of `index` becomes the parent of `index`.
+            // Therefore, `index` must become the left child of `right_index`
+            // (its value is lower than that of `right_index`). If `index`
+            // becomes the left child of `right_index`, that means the 'old'
+            // left child of `right_index` must become the new right child of
+            // `index`.
+            //
+            //       A (index)          (right_index) B
```

Comment:
I really like these ascii art diagrams :)

Reply:
Hah, thanks! Explaining the balancing algorithm would've been a lot harder without drawing